daffodil-core/src/main/scala/org/apache/daffodil/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -775,6 +782,25 @@ trait ElementBaseGrammarMixin
       }
 
       case PrimType.Boolean => { new BinaryBoolean(this) }
+
+      case PrimType.DateTime | PrimType.Date | PrimType.Time => {
+        (primType, binaryCalendarRep) match {
+          case (PrimType.DateTime, BinaryCalendarRep.BinarySeconds) => (lengthUnits, binaryNumberKnownLengthInBits) match {
+            case (LengthUnits.Bytes, 32) => new ConvertBinaryDateTimePrim(this, binaryNumberKnownLengthInBits)
```

Comment:
I would think it would be ok but I added this because it's specified in the spec - "12.3.7.2.5   Length of Base-2 Binary Calendar Elements
Calendars with binary representation and dfdl:binaryCalendarRep ‘binarySeconds’ or ‘binaryMilliseconds’ are represented as a bit string which contains a base-2 representation.The specified length must be either exactly 4 bytes or exactly 8 bytes respectively.

The dfdl:lengthUnits property must be 'bytes'. It is a schema definition error otherwise."

Reply:
Yes, these restrictions on lengthUnits turned out to be a bad idea ultimately. They require a special check, and add no value over just requiring the length to ultimately be exactly 32 or 64 bits regardless of units used to express that. When a data format expresses *everything* in units of bits, these restrictions just make it awkward to shift to bytes units for these few cases. 

But the DFDL spec does require the check.... unfortunately.